azure-kusto-ingest/azure/kusto/ingest/ingestion_properties.py

```diff
@@ -69,15 +71,112 @@ class ReportMethod(IntEnum):
     Queue = 0
 
 
+class MappingConsts(Enum):
+    """Mapping properties keys."""
+    # Json Mapping consts
+    PATH = "Path",
+    TRANSFORMATION_METHOD = "Transform"
+    # csv Mapping consts
+    ORDINAL = "Ordinal"
+    CONST_VALUE = "ConstValue"
+    # Avro Mapping consts
+    FIELD_NAME = "Field"
+    COLUMNS = "Columns"
+    # General Mapping consts
+    STORAGE_DATA_TYPE = "StorageDataType"
+
+
+class TransformationMethod(Enum):
+    # Transformations to configure over json column mapping
+    NONE = "None"
+    PROPERTY_BAG_ARRAY_TO_DICTIONARY = "PropertyBagArrayToDictionary",
+    SOURCE_LOCATION = "SourceLocation"
+    SOURCE_LINE_NUMBER = "SourceLineNumber"
+    GET_PATH_ELEMENT = "GetPathElement"
+    UNKNOWN_ERROR = "UnknownMethod"
+    DATE_TIME_FROM_UNIX_SECONDS = "DateTimeFromUnixSeconds"
+    DATE_TIME_FROM_UNIX_MILLISECONDS = "DateTimeFromUnixMilliseconds"
+    DATE_TIME_FROM_UNIX_MICROSECONDS = "DateTimeFromUnixMicroseconds"
+    DATE_TIME_FROM_UNIX_NANOSECONDS = "DateTimeFromUnixNanoseconds"
+
+
 class ColumnMapping:
-    """Abstract class to column mapping."""
+    """ Common class to all the column mapping kinds."""
+
+    def __init__(self, columnName, cslDataType, properties={}):
+        self.columnName = columnName
+        self.columnType = cslDataType
+        self.properties = properties
+
+    def setPath(self, path):
```

Comment:
In addition to the other comment - getters and setters are not very pythonic, mainly used when there is a validation logic or encapsulation, why not add optional parameters to the constructor?

Reply:
ok